### DISCUSSION TITLE
RAG not working with `gpt-3.5-turbo`

### DISCUSSION THREAD

**Discussion by s2terminal** [danny-avila/LibreChat#2373] [Troubleshooting]
I have `OPENAI_MODELS` set to `gpt-3.5-turbo` now.

```.env
OPENAI_MODELS=gpt-3.5-turbo
```

I tried to use the RAG feature added in v0.7.0. However, the AI chat replies fine, but does not search for attachments.

I changed `OPENAI_MODELS` to `gpt-3.5-turbo-0125` instead of `gpt-3.5-turbo` and the RAG works as expected, even though [the model has not changed](https://platform.openai.com/docs/models/gpt-3-5-turbo).

```.env
OPENAI_MODELS=gpt-3.5-turbo-0125
```

When I used `gpt-3.5-turbo`, the log says `maxContextTokens: 4092`. I also see in my logs that `modelOptions.messages` has only one `role: user`.

```
2024-04-10T08:45:49.851Z debug: [BaseClient] instructions tokenCount: 4687
2024-04-10T08:45:49.852Z debug: [BaseClient] Context Count (1/2)
{
  remainingContextTokens: 4078,
  maxContextTokens: 4092,
}
```

When I looked at the logs when I used `gpt-3.5-turbo-0125` and it worked, I saw `maxContextTokens: 16375` and `modelOptions.messages` contained `instructions`.

```
2024-04-10T08:46:04.738Z debug: [BaseClient] instructions tokenCount: 4687
2024-04-10T08:46:04.739Z debug: [BaseClient] Context Count (1/2)
{
  remainingContextTokens: 11461,
  maxContextTokens: 16375,
}
```

```
"role":"system","name":"instructions","content":"Instructions:\nThe user has attached a file to the... [truncated],
```

I think that the calculation of maxContextTokens when using `gpt-3.5-turbo` is incorrect. I am also uncomfortable with the fact that the chat replies correctly even though the RAG Instruction is not working correctly. (I have no good ideas to fix this)

Since `gpt-3.5-turbo` and `gpt-3.5-turbo-0125` are the same, I can avoid this problem by setting only `gpt-3.5-turbo-0125`.

**Comment by danny-avila** [comment]
I'll be updating the default limit for gpt-3.5-turbo since they updated this, and also looking into fixing this for lower context models.

> I am also uncomfortable with the fact that the chat replies correctly even though the RAG Instruction is not working correctly. (I have no good ideas to fix this)

This shouldn't be surprising with LLMs, as they can reply incorrectly when the instructions are perfect
